data/remnant/remnant missions.txt

```diff
@@ -3463,3 +3463,265 @@ event "remnant: pantica void sprites"
+			`	Just as you arrive and reach for the console to open the airlock, you sense a change in your surroundings. Looking around, you see that a group of Remnant gathered around the two of you, focused on Dawn. A young Remnant - scarcely more than a child - steps forward. "Thank you for your patience." He says, then steps back. An elderly Remnant steps forward. "Thank you for your curiosity." She says, then steps back. A middle-aged Remnant steps forward. "Go with our blessing, Seeker, and know that wherever you go, the Remnant go with you." She says it simply, but you feel the weight of tradition behind them. She turns to you. "And you, stranger who walks the stars... May the Embers guide your steps as you walk among us." She nods respectfully, then steps back.`
+			`	As soon as the third steps back there is a flurry of movement as each of the Remnant present quickly step forward, give her a quick hug, and then scurry off into the starport. Before you process what is going on all the Remnant gathered around you have vanished. You look at Dawn, and her expression is a mixture of awe and humility.`
+			choice
+				`	"They know you are coming back, right?"`
+				`	"What was that about?"`
+				`	"Alright, let's get going."`
+					goto astralmoving
+			`	"Well, what we are doing is dangerous. We will be crossing Korath territory, a wide expanse of uninhabited space, and then scanning a black hole. I have confidence that we will return, but nonetheless there is still a chance that we die in the process.`
+			`	"Long ago we, as a people, decided that we must face the fear of separation, the unknown, and the danger. One of the ways we do that is by holding these informal ceremonies. When one of us is going to depart on their first long distance voyage into hostile territory, a group of Remnant gather to bid them farewell. It is a way to strengthen ourselves and remind each other of our connections."`
+			choice
+				`	"Thanks for the explanation."`
+					goto astralmoving
+				`	"Kind of like a funeral, with friends and family gathering to see you off?"`
+			`	"Oh, no. I said my farewells to my friends and relatives before we left. Those who gathered to see me off today were more of a representation of Remnant society. I only recognized a few of them, but they took the time to come down to the <ship> to see me before I left. They came to see me, and wish me well, so that no matter what challenges I face, I will know that the Remnant remember me and appreicate what I am doing." Her gestures fade into contemplative stillness for a moment, then resume motion.`
+			label astralmoving
+			`	She gestures towards the airlock door. "Lead the way, Captain."`
+				accept
+	on enter "Sagittarius A*"
```

Comment:
Is it possible to specifyconditions for this, like a `to waypoint` in the same way that we can specify conditions for, say, an `on complete` ? Specifically, asking if it is possible to check here to ensure that the player does actually have the "Research Laboratory" on their ship, and didn't just ditch it after they accepted the mission?

(I realize I could specify taking off the "Research Laboratory" to be a fail condition, but I don't want to hinder people's outfitting or fail the mission - I just don't want them to be able to complete the waypoint without it.)

Reply:
`require` still goes in `on *` blocks, so theoretically one could prevent the `on waypoint` action from triggering if one didn't have a specific outfit, *but* the waypoint objective would still be complete for that system, meaning that you can prevent the `on waypoint` from showing, but if it doesn't show and the player returns with the correct outfits, it won't then trigger the `on waypoint` because the waypoint objective was already complete.

Reply:
hmmm... That's problematic.

Reply:
Just don't worry about it. The mission requires that the player have one at the start. If they're really going to go through the motions of having the outfit installed in order to start the mission then removing it after accepting the mission, then let them have at it. We can't stop them from doing that, so it's on them that the story doesn't make sense without a lab installed.

Reply:
If you *really* wanted to require that the player have a lab installed, try putting `require "Research Laboratory"` in the `on enter`, then have the `on enter` also set a condition that is required for mission completion. This way the `on enter` won't trigger unless the lab is present, and the mission can't be completed unless the `on enter` is triggered. What you couldn't do though is have a dialog pop up if the player arrives without a lab.

Reply:
That's true. I'll just leave this as is. If the player *really* wants to jump through hoops to do the mission without a lab...